8319192: Remove javax.swing.plaf.synth.SynthLookAndFeel.load(URL url)

prrace opened the pull request:
Back in JDK 21 we deprecated-for-removal SynthLookAndFeel.load(URL) and targeted it to be removed under this bug ID in JDK 25 .. which is now here.

No open tests needed to be updated to accommodate this removal 
I did find one closed manual test, written to explicitly test this API back when it was added, and I'll need to remove that, but it doesn't affect this PR.

There will be a CSR (and probably a release note too).

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8351238](https://bugs.openjdk.org/browse/JDK-8351238) to be approved

### Issues
 * [JDK-8319192](https://bugs.openjdk.org/browse/JDK-8319192): Remove javax.swing.plaf.synth.SynthLookAndFeel.load(URL url) (**Bug** - P4)
 * [JDK-8351238](https://bugs.openjdk.org/browse/JDK-8351238): Remove javax.swing.plaf.synth.SynthLookAndFeel.load(URL url) (**CSR**)


### Reviewers
 * [Alexander Zvegintsev](https://openjdk.org/census#azvegint) (@azvegint - **Reviewer**)
 * [Tejesh R](https://openjdk.org/census#tr) (@TejeshR13 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/23909/head:pull/23909` \
`$ git checkout pull/23909`

Update a local copy of the PR: \
`$ git checkout pull/23909` \
`$ git pull https://git.openjdk.org/jdk.git pull/23909/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 23909`

View PR using the GUI difftool: \
`$ git pr show -t 23909`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/23909.diff">https://git.openjdk.org/jdk/pull/23909.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/23909#issuecomment-2699301104)
</details>

prrace commented:
/csr

prsadhuk commented on src/java.desktop/share/classes/javax/swing/plaf/synth/SynthParser.java:
```diff
@@ -255,17 +246,7 @@ public void parse(InputStream inputStream,
-                return null;
-            }
-        }
+        return _classResourceBase.getResource(path);
```
Any particular reason for removing the null check?

prrace replied:
Yes. the resource was *either* url-based or classpath based. One would be null, the other not.
Now there's only one, so it must be the non-null one.

azvegint commented:
It is also mentioned in https://github.com/openjdk/jdk/blob/caaf4098452476d981183ad4302b76b9c883a72b/src/java.desktop/share/classes/javax/swing/plaf/synth/doc-files/synthFileFormat.html#L57-L58

azvegint commented on src/java.desktop/share/classes/javax/swing/plaf/synth/SynthParser.java:
```diff
@@ -213,26 +207,23 @@ class SynthParser extends DefaultHandler {
         }
 
-        assert(!(urlResourceBase != null && classResourceBase != null));
+        assert(classResourceBase != null);
```
It looks like this assert is no longer needed, because the `classResourceBase` is not null at this point.

prrace replied:
Fair enough. But I'm not sure it was ever needed. The condition in the existing code also ensured it would never fail.

prrace replied:
fixed

azvegint replied:
> But I'm not sure it was ever needed.

It seems that this was a check that both `urlResourceBase` and `classResourceBase` are not null at the same time, but only when assertions are enabled.
Not sure why it was an assertion.

prrace commented:
> > It is also mentioned in
> > https://github.com/openjdk/jdk/blob/caaf4098452476d981183ad4302b76b9c883a72b/src/java.desktop/share/classes/javax/swing/plaf/synth/doc-files/synthFileFormat.html#L57-L58
> 
> Ah. I'd forgotten that. I had updated it back in JDK 21 but it slipped my mind this time round.

fixed

mrserb commented on src/java.desktop/share/classes/javax/swing/plaf/synth/SynthParser.java:
```diff
@@ -213,26 +207,21 @@ class SynthParser extends DefaultHandler {
      *
      * @param inputStream XML document containing the styles to read
      * @param factory DefaultSynthStyleFactory that new styles are added to
-     * @param urlResourceBase the URL used to resolve any resources, such as Images
```
The description of this method includes information about the URL

prrace replied:
The whole sentence is now unnecessary and can be deleted.

mrserb commented on src/java.desktop/share/classes/javax/swing/plaf/synth/SynthParser.java:
```diff
@@ -34,7 +34,6 @@
 import java.io.InputStream;
-import java.net.MalformedURLException;
 import java.net.URL;
 import java.net.URLClassLoader;
```
Is URLClassLoader no longer used?

prrace replied:
seems so. I'll remove it.

prsadhuk replied:
URLCLassLoader still not removed it seems..I dont see its usage in the file..

TejeshR13 reviewed (approved):
LGTM. Just have to update copyright year.

prrace commented:
/integrate
